### DISCUSSION TITLE
Changing the Edge Color of the molecule in RDKit.MolDraw2D

### DISCUSSION THREAD

**Discussion by rururudolph145** [rdkit/rdkit#5977] [Q&A]
Hello,

I have been trying to draw a molecule with several highlights on a specific substructure.

By using highlightAtomColors I can change the color of molecule that I want to highlight like the image below.

<img width="548" alt="image" src="https://user-images.githubusercontent.com/20742273/212474815-48d65774-0653-4a0d-b9b8-9e283097a9fb.png">

My question is, is there an option where I can change the edge color/styles of the highlight as well? For example, I want to put a black border over the highlighted atom. Or maybe a dashed border over the highlighted atom just like the image below?
<img width="548" alt="image" src="https://user-images.githubusercontent.com/20742273/212474769-d991a4ee-ae2c-4a0f-8099-569eaf032624.png">

Thank you in advance.

**Comment by DavidACosgrove** [comment]
There is nothing in the drawing code to do that sort of thing
thing directly, and it would be quite a lot of work to introduce it.  It is
possible to add arbitrary shapes to the image, however, so with a bit of
work you could achieve the effect.  At the bottom of this discussion
https://github.com/rdkit/rdkit/discussions/4832 there is some code that
puts arcs on the drawing to show bond angles, and something similar could
be used to put circles round the highlights in your case.  You could
probably change the style of the line as well.

Dave


On Sat, Jan 14, 2023 at 1:41 PM Rudolf Jason ***@***.***>
wrote:

> Hello,
>
> I have been trying to draw a molecule with several highlights on a
> specific substructure.
>
> By using highlightAtomColors I can change the color of molecule that I
> want to highlight like the image below.
>
> [image: image]
> <https://user-images.githubusercontent.com/20742273/212474455-2b802ce7-15c8-4826-a80d-a4aa98609485.png>
>
> My question is, is there an option where I can change the edge
> color/styles of the highlight as well? For example, I want to put a black
> border over the highlighted atom. Or maybe a dashed border over the
> highlighted atom just like the image below?
> [image: image]
> <https://user-images.githubusercontent.com/20742273/212474769-d991a4ee-ae2c-4a0f-8099-569eaf032624.png>
>
> Thank you in advance.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/5977>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FX75PHQKPVOHIMR7XTWSKUIZANCNFSM6AAAAAAT3JBFPI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk
